scripts/run.py

```diff
@@ -0,0 +1,487 @@
+
+class PipelineImpl:
+    def __init__(self, config_file):
+        self.logger = Logger()
+        self.config = Config(config_file, self.logger)
+
+    def local(self, command, directory, verbose=False):
+        with lcd(directory):
+            res = local(command, capture=True)
+            if verbose:
+                self.logger.raw(str(res.stdout), str(res.stderr))
+            return res
+
+    def local_canfail(self, name, command, directory, verbose=False, interactive=False):
+        failed = False
+        with settings(warn_only=True):
+            result = self.local(command, directory, verbose=verbose)
+            if result.failed:
+                with lcd(directory):
+                    with open(os.path.join(directory, name.replace(" ", "_")), 'w') as command_report:
+                        command_report.write(result.stdout)
+                        command_report.write(result.stderr)
+                if not interactive:
+                    failed = True
+                else:
+                    if confirm("%s failed. Abort?" % name):
+                        abort("Aborting at user request.")
+                    else:
+                        failed = True
+        return failed
+
+    def load_project_info(self, path):
+        info_path = os.path.join(path, "project.csv")
+        if not os.path.exists(info_path):
+            create_project_info(path)
+        with open(info_path) as csv_file:
+            reader = csv.reader(csv_file)
+            names = next(reader)
+            data = next(reader)
+            info = dict(zip(names, data))
+            self.info("[%s] Load project info" % info["name"])
+            return info
+
+    def get_report(self, project_path, kind):
+        report_path = os.path.join(project_path, self.config.get(kind))
+        for root, subdir, files in os.walk(project_path):
+            for f in files:
+                if f == self.config.get(kind):
+                    with open(os.path.join(root, f)) as report:
+                        return report.read()
+
+        return None
+
+    def write_report(self, content, project_path, kind):
+        report_path = os.path.join(project_path, self.config.get(kind))
+        with open(report_path, 'w') as report:
+            return report.write(content)
+
+    def info(self, msg):
+        if self.config.get("debug_info"):
+            self.logger.log(msg)
+
+    def error(self, msg):
+        if self.config.get("debug_info"):
+            self.logger.error(msg)
+
+    def checkout_latest_tag(self, project_name, project_path):
+        self.info("[Import][%s] Checkout latest tag..." % project_name)
+        self.local("git fetch --tags --depth 1", project_path)
+        self.local_canfail("Load latest tag", "latestTag=$( git describe --tags `git rev-list --tags --max-count=1` )", directory=project_path)
+        self.local("git checkout $latestTag", project_path)
+
+    def has_file_with_ending(self, folder, extension):
```

Comment:
Not sure I understand it. I think we should worry only about files in root of the projects and so I would go for:
```python
os.path.exists(os.path.join(project_root, file))
```

What we will have to do eventually is configure custom build handlers for different projects.
For example, you can have projects that are built like:
- `mvn -f releng/build.sbt`
- `./build.sh`